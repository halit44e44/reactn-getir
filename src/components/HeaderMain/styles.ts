import {StyleSheet, Dimensions} from "react-native";

const {height} = Dimensions.get('window')

const styles = StyleSheet.create({
    headerMain: {
        height: height * 0.064,
        backgroundColor: "#F7D102"
    },
    headerOne: {
        height: height * 0.064 ,
        width: "80%",
        backgroundColor: "#FFF",
        flexDirection: "row",
        alignItems: "center",
        borderBottomRightRadius: height * 0.032,
        borderTopRightRadius: height * 0.032,
    },
    headerOneView: {
        flexDirection: "row",
        alignItems: "center",
        borderLeftWidth: 2,
        borderLeftColor: "#F3F2FD",
        paddingLeft: 10
    },
    headerImage: {
        width: 30,
        height: 30,
        margin: 10
    },

    headerTwo: {
        width: "25%",
        height: height * 0.064,
        left: 30,
        justifyContent: "center",
        alignItems: "center"
    }
})

export default styles;
