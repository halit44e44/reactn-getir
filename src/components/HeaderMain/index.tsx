import React from 'react';
import {View, Text, Image} from "react-native";
import styles from "./styles";
import {Entypo} from '@expo/vector-icons';

const Index = () => {
    return (
        <View style={styles.headerMain}>
            <View style={styles.headerOne}>
                <Image style={styles.headerImage} source={{uri: "https://cdn.getir.com/misc/emoji/house.png"}}/>
                <View style={styles.headerOneView}>
                    <Text style={{fontWeight: "600", fontSize: 16}}>Ev</Text>
                    <Text style={{fontWeight: "500", fontSize: 11.5, color: "gray", marginLeft: 6, marginRight: 3}}>Dedepaşa
                        Blv. Yenişehir Mahallesi.. </Text>
                    <Entypo testID="right-icon" name="chevron-right" size={22} color="#5D3EBD"/>
                </View>
                <View style={styles.headerTwo}>
                    <Text style={{fontSize: 10, fontWeight: "bold", color: "#5D3EBD"}}>TVS</Text>
                    <Text style={{fontSize: 20, fontWeight: "bold", color: "#5D3EBD"}}>13<Text style={{ fontSize: 16 }}>dk</Text></Text>
                </View>
            </View>
            <View>
            </View>
        </View>
    );
};

export default Index;
