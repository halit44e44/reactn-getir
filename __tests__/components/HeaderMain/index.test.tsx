import React from "react";
import {render} from "@testing-library/react-native";
import HeaderMain from "../../../src/components/HeaderMain";


describe("components/HeaderMain", () => {
    test("renders Correctly", () => {
        const {toJSON} = render(<HeaderMain/>)
        expect(toJSON()).toMatchSnapshot()
    })
    test("renders ev text correctly", () => {
        const {getByTestId, getByText} = render(<HeaderMain/>)
        const ev = getByText('Ev')
        expect(ev).toBeDefined()
    })
    test("render the correct right icon", () => {
        const {getByTestId, getByText} = render(<HeaderMain/>)
        const icon = getByTestId('right-icon')
        expect(icon.props.style[2].fontFamily).toEqual('entypo')
        expect(icon.props.style[0].fontSize).toEqual(22)
    })
})
