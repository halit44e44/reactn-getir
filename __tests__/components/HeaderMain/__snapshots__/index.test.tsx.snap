// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/HeaderMain renders Correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#F7D102",
      "height": 85.376,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderBottomRightRadius": 42.688,
        "borderTopRightRadius": 42.688,
        "flexDirection": "row",
        "height": 85.376,
        "width": "80%",
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://cdn.getir.com/misc/emoji/house.png",
        }
      }
      style={
        {
          "height": 30,
          "margin": 10,
          "width": 30,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "borderLeftColor": "#F3F2FD",
          "borderLeftWidth": 2,
          "flexDirection": "row",
          "paddingLeft": 10,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Ev
      </Text>
      <Text
        style={
          {
            "color": "gray",
            "fontSize": 11.5,
            "fontWeight": "500",
            "marginLeft": 6,
            "marginRight": 3,
          }
        }
      >
        Dedepaşa Blv. Yenişehir Mahallesi.. 
      </Text>
      <Text />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "height": 85.376,
          "justifyContent": "center",
          "left": 30,
          "width": "25%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#5D3EBD",
            "fontSize": 10,
            "fontWeight": "bold",
          }
        }
      >
        TVS
      </Text>
      <Text
        style={
          {
            "color": "#5D3EBD",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        13
        <Text
          style={
            {
              "fontSize": 16,
            }
          }
        >
          dk
        </Text>
      </Text>
    </View>
  </View>
  <View />
</View>
`;
